pkg/registry/apis/provisioning/repository/github/client.go

```diff
@@ -0,0 +1,108 @@
+	// When a file is checked, the first returned value will have a value. For a directory, the second will. The other value is always nil.
+	// If an error occurs, the returned values may or may not be nil.
+	//
+	// If ".." appears in the "path", this method will return an error.
+	GetContents(ctx context.Context, owner, repository, path, ref string) (fileContents RepositoryContent, dirContents []RepositoryContent, err error)
+
+	// CreateFile creates a new file in the repository under the given path.
+	// The file is created on the branch given.
+	// The message given is the commit message. If none is given, an appropriate default is used.
+	// The content is what the file should contain. An empty slice is valid, though often not very useful.
+	//
+	// If ".." appears in the "path", this method will return an error.
+	CreateFile(ctx context.Context, owner, repository, path, branch, message string, content []byte) error
+
+	// UpdateFile updates a file in the repository under the given path.
+	// The file is updated on the branch given.
+	// The message given is the commit message. If none is given, an appropriate default is used.
+	// The content is what the file should contain. An empty slice is valid, though often not very useful.
+	// If the path does not exist, an error is returned.
+	// The hash given must be the SHA hash of the file contents. Calling GetContents in advance is an easy way of handling this.
+	//
+	// If ".." appears in the "path", this method will return an error.
+	UpdateFile(ctx context.Context, owner, repository, path, branch, message, hash string, content []byte) error
+
+	// DeleteFile deletes a file in the repository under the given path.
+	// The file is deleted from the branch given.
+	// The message given is the commit message. If none is given, an appropriate default is used.
+	// If the path does not exist, an error is returned.
+	// The hash given must be the SHA hash of the file contents. Calling GetContents in advance is an easy way of handling this.
+	//
+	// If ".." appears in the "path", this method will return an error.
+	DeleteFile(ctx context.Context, owner, repository, path, branch, message, hash string) error
+
+	ListWebhooks(ctx context.Context, owner, repository string) ([]WebhookConfig, error)
+	CreateWebhook(ctx context.Context, owner, repository string, cfg WebhookConfig) error
+	DeleteWebhook(ctx context.Context, owner, repository string, webhookID int64) error
+	EditWebhook(ctx context.Context, owner, repository string, cfg WebhookConfig) error
+}
+
+type RepositoryContent interface {
+	// Returns true if this is a directory, false if it is a file.
+	IsDirectory() bool
+	// Returns the contents of the file. Decoding happens if necessary.
+	// Returns an error if the content represents a directory.
+	GetFileContent() (string, error)
+	// Returns true if this is a symlink.
+	// If true, GetPath returns the path where this symlink leads.
+	IsSymlink() bool
```

Comment:
The intent is to avoid using these files eventually, as they may result in security risks.